### PR TITLE
fix(PersonalInfoTab && InvestigatedPersonInfo): initialize subOccupation on occupation change and in process sub status fixes

### PR DESCRIPTION
**Pull request by roymaniv**

### CODE REVIEW

File: client/src/Utils/auxiliaryFunctions/auxiliaryFunctions.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ const isIdValid = (id: string | null | undefined) => {
       sum += digitMul;
     })
   }
-  return sum % 10 === 0;
+  return sum % 10 === 0 && sum > 0;
```
Comment (daniel7764) [inline client/src/Utils/auxiliaryFunctions/auxiliaryFunctions.ts:21]: weird, that's a fix i already pushed to dev

File: client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx [hunk 2]
```diff
@@ -110,6 +109,12 @@ const PersonalInfoTab: React.FC<Props> = ({ id }: Props): JSX.Element => {
         }
     }
 
+    useEffect(() => {
+        if (subOccupationName === '') {
```
Comment (daniel7764) [inline client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx:113]: you can do !Boolean(subOccupationName)

File: client/src/Utils/auxiliaryFunctions/auxiliaryFunctions.ts (hunk 1 above)
Comment (roymaniv) [reply client/src/Utils/auxiliaryFunctions/auxiliaryFunctions.ts:21]: did you push it to test?

Comment (daniel7764) [reply client/src/Utils/auxiliaryFunctions/auxiliaryFunctions.ts:21]: yes

**Review by ItayBenMoshe — APPROVED** [review]

**Review by nevobonneh — APPROVED** [review]

**Pull request merged**